### PR TITLE
Add match media change listener in browser ts

### PR DESCRIPTION
**Pull request by a-stewart**
This PR fixes #143197

Logic based on if the window is standalone should be based on if it is currently standalone, rather than if it was standalone when VS Code was started.

This change also moves `addMatchMediaChangeListener` from browser.ts to dom.ts so that it can be used in dom.ts without creating a circular dependency.

### CODE REVIEW

File: src/vs/base/browser/dom.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ import * as platform from 'vs/base/common/platform';
 import { withNullAsUndefined } from 'vs/base/common/types';
 import { URI } from 'vs/base/common/uri';
 
+export { addMatchMediaChangeListener } from 'vs/base/browser/browser';
```
Comment (bpasero) [inline src/vs/base/browser/dom.ts:21]: Why is this being re-exported?

Comment (bpasero) [reply src/vs/base/browser/dom.ts:21]: Yes why not.

Comment (a-stewart) [reply src/vs/base/browser/dom.ts]: Done, sorry for the delay

**Review by bpasero — APPROVED** [review]
LGTM. Arguably even though `isStandalone` is now dynamic, there are still a few places where we statically call this method on startup and then fail to reconcile the change. But I think your scenario should be covered.

**Pull request merged**